include/gillwald/geometry.hpp

```diff
@@ -6,13 +6,42 @@
 #include <vector>
 
 namespace geometry {
+
+  /**
+   * @brief Creates a Cell structure that contains X,Y coordinate
+   * @param x X coordinate
+   * @param y Y coordinate
+   */	
+	struct Cell {
+	  Cell(int x, int y) : x(x),y(y) {};
+	  int x; ///< The x coordinate of the cell
+	  int y; ///< The y coordinate of the cell
+	};
+
   /**
    * @brief Creates discretized line between two end points
    * @param x0 X coordinate of starting point
    * @param y0 Y coordinate of starting point
    * @param x1 X coordinate of ending point
    * @param y1 Y coordinate of ending point
    * @returns Pixel coordinates representing line
-   */
+   */	
   std::vector<std::vector<int>> bresenham_conversion(int x0, int y0, int x1, int y1);
+
+  /**
+   * @brief Creates discretized line between two end points
+   * @param start The starting coordinate of the line
+   * @param end The ending coordinate of the line
+   * @returns Pixel coordinates representing line
+   */	
+  std::vector<std::vector<int>> bresenham_conversion(const Cell &start, const Cell &end);
+
+  /**
+   * @brief Creates discretized line between two end points
+   * @param start The starting coordinate of the line
+   * @param end The ending coordinate of the line
+   * @param max_ray_length The maximum number of elements the returned vector should contain
+   * @returns Pixel coordinates representing line
+   */	  
+  [[nodiscard]] std::vector<Cell> raytraceline_conversion(const Cell& start, const Cell& end, size_t max_ray_length);
```

Comment:
Can we just call this `raytrace`? As in `[[nodiscard]] std::vector<Cell> raytrace(const Cell& ...)`
I'm not really sure why the `line_conversion` is needed.

Reply:
`max_ray_length` is too long of a name. `max_length` should do

Reply:
I thought names were supposed to be more descriptive. But I'll change it.

Reply:
Solution is to check if size is too big, or negative, and return an empty vector.

Reply:
Done